File: google_fit_test_task/lib/measurmnetObject/measurement.dart
import 'fieldMeasurements.dart';

class Measurement {
  String? type;
  String? dateStart;
  String? dateEnd;
  List<FieldMeasurements>? fieldMeasurements;

  Measurement(
      {this.type, this.dateStart, this.dateEnd, this.fieldMeasurements});

  Measurement.fromJson(Map<String, dynamic> json) {
    type = json['type'];
    dateStart = json['dateStart'];
    dateEnd = json['dateEnd'];

    if (json['FieldMeasurements'] != null) {
      fieldMeasurements = <FieldMeasurements>[];
      json['FieldMeasurements'].forEach((v) {
        fieldMeasurements!.add(FieldMeasurements.fromJson(v));
      });
    }
    
  }

  


}

File: google_fit_test_task/android/app/src/main/kotlin/com/example/google_fit_test_task/Measurement.kt
package com.example.google_fit_test_task

import com.example.google_fit_test_task.FieldMeasurement
import kotlinx.serialization.Serializable

@Serializable
data class Measurement(val type: String, val dateStart: String, val dateEnd: String, val FieldMeasurements : ArrayList<FieldMeasurement>)

 // var value = _value
